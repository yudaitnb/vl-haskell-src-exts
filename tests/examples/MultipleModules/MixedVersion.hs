main =
  let x = unversion
            (version A = 1.0.0 of f 1)
      y = unversion
            (version A = 2.0.0 of f 1)
  in x + y