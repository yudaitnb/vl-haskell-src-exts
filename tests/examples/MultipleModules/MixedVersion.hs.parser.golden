ParseOk
  ( Module
      SrcSpanInfo
        { srcInfoSpan =
            SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 1 1 7 1
        , srcInfoPoints =
            [ SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 1 1 1 1
            , SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 1 1 1 1
            , SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 1 1 1 1
            , SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 7 1 7 1
            , SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 7 1 7 1
            ]
        }
      Nothing
      []
      []
      [ PatBind
          SrcSpanInfo
            { srcInfoSpan =
                SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 1 1 6 11
            , srcInfoPoints = []
            }
          (PVar
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 1 1 1 5
               , srcInfoPoints = []
               }
             (Ident
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 1 1 1 5
                  , srcInfoPoints = []
                  }
                "main"))
          (UnGuardedRhs
             SrcSpanInfo
               { srcInfoSpan =
                   SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 1 6 6 11
               , srcInfoPoints =
                   [ SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 1 6 1 7
                   ]
               }
             (Let
                SrcSpanInfo
                  { srcInfoSpan =
                      SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 2 3 6 11
                  , srcInfoPoints =
                      [ SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 2 3 2 6
                      , SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 3 6 5
                      ]
                  }
                (BDecls
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 2 7 5 39
                     , srcInfoPoints =
                         [ SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 2 7 2 7
                         , SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 4 7 4 7
                         , SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 3 6 0
                         ]
                     }
                   [ PatBind
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 2 7 3 39
                         , srcInfoPoints = []
                         }
                       (PVar
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 2 7 2 8
                            , srcInfoPoints = []
                            }
                          (Ident
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 2 7 2 8
                               , srcInfoPoints = []
                               }
                             "x"))
                       (UnGuardedRhs
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 2 9 3 39
                            , srcInfoPoints =
                                [ SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 2 9 2 10
                                ]
                            }
                          (VExt
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan
                                     "tests/examples/MultipleModules/MixedVersion.hs" 2 11 3 39
                               , srcInfoPoints = []
                               }
                             (Paren
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan
                                        "tests/examples/MultipleModules/MixedVersion.hs" 3 13 3 39
                                  , srcInfoPoints =
                                      [ SrcSpan
                                          "tests/examples/MultipleModules/MixedVersion.hs" 3 13 3 14
                                      , SrcSpan
                                          "tests/examples/MultipleModules/MixedVersion.hs" 3 38 3 39
                                      ]
                                  }
                                (VRes
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan
                                           "tests/examples/MultipleModules/MixedVersion.hs"
                                           3
                                           14
                                           3
                                           38
                                     , srcInfoPoints =
                                         [ SrcSpan
                                             "tests/examples/MultipleModules/MixedVersion.hs"
                                             3
                                             14
                                             3
                                             21
                                         , SrcSpan
                                             "tests/examples/MultipleModules/MixedVersion.hs"
                                             3
                                             32
                                             3
                                             34
                                         ]
                                     }
                                   (VBinds
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan
                                              "tests/examples/MultipleModules/MixedVersion.hs"
                                              3
                                              22
                                              3
                                              31
                                        , srcInfoPoints =
                                            [ SrcSpan
                                                "tests/examples/MultipleModules/MixedVersion.hs"
                                                3
                                                22
                                                3
                                                22
                                            , SrcSpan
                                                "tests/examples/MultipleModules/MixedVersion.hs"
                                                3
                                                32
                                                3
                                                32
                                            ]
                                        }
                                      [ VBind
                                          SrcSpanInfo
                                            { srcInfoSpan =
                                                SrcSpan
                                                  "tests/examples/MultipleModules/MixedVersion.hs"
                                                  3
                                                  22
                                                  3
                                                  31
                                            , srcInfoPoints =
                                                [ SrcSpan
                                                    "tests/examples/MultipleModules/MixedVersion.hs"
                                                    3
                                                    24
                                                    3
                                                    25
                                                ]
                                            }
                                          (ModuleName
                                             SrcSpanInfo
                                               { srcInfoSpan =
                                                   SrcSpan
                                                     "tests/examples/MultipleModules/MixedVersion.hs"
                                                     3
                                                     22
                                                     3
                                                     23
                                               , srcInfoPoints = []
                                               }
                                             "A")
                                          (VersionNumber
                                             SrcSpanInfo
                                               { srcInfoSpan =
                                                   SrcSpan
                                                     "tests/examples/MultipleModules/MixedVersion.hs"
                                                     3
                                                     26
                                                     3
                                                     31
                                               , srcInfoPoints = []
                                               }
                                             1
                                             0
                                             0)
                                      ])
                                   (App
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan
                                              "tests/examples/MultipleModules/MixedVersion.hs"
                                              3
                                              35
                                              3
                                              38
                                        , srcInfoPoints = []
                                        }
                                      (Var
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan
                                                 "tests/examples/MultipleModules/MixedVersion.hs"
                                                 3
                                                 35
                                                 3
                                                 36
                                           , srcInfoPoints = []
                                           }
                                         (UnQual
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan
                                                    "tests/examples/MultipleModules/MixedVersion.hs"
                                                    3
                                                    35
                                                    3
                                                    36
                                              , srcInfoPoints = []
                                              }
                                            (Ident
                                               SrcSpanInfo
                                                 { srcInfoSpan =
                                                     SrcSpan
                                                       "tests/examples/MultipleModules/MixedVersion.hs"
                                                       3
                                                       35
                                                       3
                                                       36
                                                 , srcInfoPoints = []
                                                 }
                                               "f")))
                                      (Lit
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan
                                                 "tests/examples/MultipleModules/MixedVersion.hs"
                                                 3
                                                 37
                                                 3
                                                 38
                                           , srcInfoPoints = []
                                           }
                                         (Int
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan
                                                    "tests/examples/MultipleModules/MixedVersion.hs"
                                                    3
                                                    37
                                                    3
                                                    38
                                              , srcInfoPoints = []
                                              }
                                            1
                                            "1")))))))
                       Nothing
                   , PatBind
                       SrcSpanInfo
                         { srcInfoSpan =
                             SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 4 7 5 39
                         , srcInfoPoints = []
                         }
                       (PVar
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 4 7 4 8
                            , srcInfoPoints = []
                            }
                          (Ident
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 4 7 4 8
                               , srcInfoPoints = []
                               }
                             "y"))
                       (UnGuardedRhs
                          SrcSpanInfo
                            { srcInfoSpan =
                                SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 4 9 5 39
                            , srcInfoPoints =
                                [ SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 4 9 4 10
                                ]
                            }
                          (VExt
                             SrcSpanInfo
                               { srcInfoSpan =
                                   SrcSpan
                                     "tests/examples/MultipleModules/MixedVersion.hs" 4 11 5 39
                               , srcInfoPoints = []
                               }
                             (Paren
                                SrcSpanInfo
                                  { srcInfoSpan =
                                      SrcSpan
                                        "tests/examples/MultipleModules/MixedVersion.hs" 5 13 5 39
                                  , srcInfoPoints =
                                      [ SrcSpan
                                          "tests/examples/MultipleModules/MixedVersion.hs" 5 13 5 14
                                      , SrcSpan
                                          "tests/examples/MultipleModules/MixedVersion.hs" 5 38 5 39
                                      ]
                                  }
                                (VRes
                                   SrcSpanInfo
                                     { srcInfoSpan =
                                         SrcSpan
                                           "tests/examples/MultipleModules/MixedVersion.hs"
                                           5
                                           14
                                           5
                                           38
                                     , srcInfoPoints =
                                         [ SrcSpan
                                             "tests/examples/MultipleModules/MixedVersion.hs"
                                             5
                                             14
                                             5
                                             21
                                         , SrcSpan
                                             "tests/examples/MultipleModules/MixedVersion.hs"
                                             5
                                             32
                                             5
                                             34
                                         ]
                                     }
                                   (VBinds
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan
                                              "tests/examples/MultipleModules/MixedVersion.hs"
                                              5
                                              22
                                              5
                                              31
                                        , srcInfoPoints =
                                            [ SrcSpan
                                                "tests/examples/MultipleModules/MixedVersion.hs"
                                                5
                                                22
                                                5
                                                22
                                            , SrcSpan
                                                "tests/examples/MultipleModules/MixedVersion.hs"
                                                5
                                                32
                                                5
                                                32
                                            ]
                                        }
                                      [ VBind
                                          SrcSpanInfo
                                            { srcInfoSpan =
                                                SrcSpan
                                                  "tests/examples/MultipleModules/MixedVersion.hs"
                                                  5
                                                  22
                                                  5
                                                  31
                                            , srcInfoPoints =
                                                [ SrcSpan
                                                    "tests/examples/MultipleModules/MixedVersion.hs"
                                                    5
                                                    24
                                                    5
                                                    25
                                                ]
                                            }
                                          (ModuleName
                                             SrcSpanInfo
                                               { srcInfoSpan =
                                                   SrcSpan
                                                     "tests/examples/MultipleModules/MixedVersion.hs"
                                                     5
                                                     22
                                                     5
                                                     23
                                               , srcInfoPoints = []
                                               }
                                             "A")
                                          (VersionNumber
                                             SrcSpanInfo
                                               { srcInfoSpan =
                                                   SrcSpan
                                                     "tests/examples/MultipleModules/MixedVersion.hs"
                                                     5
                                                     26
                                                     5
                                                     31
                                               , srcInfoPoints = []
                                               }
                                             2
                                             0
                                             0)
                                      ])
                                   (App
                                      SrcSpanInfo
                                        { srcInfoSpan =
                                            SrcSpan
                                              "tests/examples/MultipleModules/MixedVersion.hs"
                                              5
                                              35
                                              5
                                              38
                                        , srcInfoPoints = []
                                        }
                                      (Var
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan
                                                 "tests/examples/MultipleModules/MixedVersion.hs"
                                                 5
                                                 35
                                                 5
                                                 36
                                           , srcInfoPoints = []
                                           }
                                         (UnQual
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan
                                                    "tests/examples/MultipleModules/MixedVersion.hs"
                                                    5
                                                    35
                                                    5
                                                    36
                                              , srcInfoPoints = []
                                              }
                                            (Ident
                                               SrcSpanInfo
                                                 { srcInfoSpan =
                                                     SrcSpan
                                                       "tests/examples/MultipleModules/MixedVersion.hs"
                                                       5
                                                       35
                                                       5
                                                       36
                                                 , srcInfoPoints = []
                                                 }
                                               "f")))
                                      (Lit
                                         SrcSpanInfo
                                           { srcInfoSpan =
                                               SrcSpan
                                                 "tests/examples/MultipleModules/MixedVersion.hs"
                                                 5
                                                 37
                                                 5
                                                 38
                                           , srcInfoPoints = []
                                           }
                                         (Int
                                            SrcSpanInfo
                                              { srcInfoSpan =
                                                  SrcSpan
                                                    "tests/examples/MultipleModules/MixedVersion.hs"
                                                    5
                                                    37
                                                    5
                                                    38
                                              , srcInfoPoints = []
                                              }
                                            1
                                            "1")))))))
                       Nothing
                   ])
                (InfixApp
                   SrcSpanInfo
                     { srcInfoSpan =
                         SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 6 6 11
                     , srcInfoPoints = []
                     }
                   (Var
                      SrcSpanInfo
                        { srcInfoSpan =
                            SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 6 6 7
                        , srcInfoPoints = []
                        }
                      (UnQual
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 6 6 7
                           , srcInfoPoints = []
                           }
                         (Ident
                            SrcSpanInfo
                              { srcInfoSpan =
                                  SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 6 6 7
                              , srcInfoPoints = []
                              }
                            "x")))
                   (QVarOp
                      SrcSpanInfo
                        { srcInfoSpan =
                            SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 8 6 9
                        , srcInfoPoints = []
                        }
                      (UnQual
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 8 6 9
                           , srcInfoPoints = []
                           }
                         (Symbol
                            SrcSpanInfo
                              { srcInfoSpan =
                                  SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 8 6 9
                              , srcInfoPoints = []
                              }
                            "+")))
                   (Var
                      SrcSpanInfo
                        { srcInfoSpan =
                            SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 10 6 11
                        , srcInfoPoints = []
                        }
                      (UnQual
                         SrcSpanInfo
                           { srcInfoSpan =
                               SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 10 6 11
                           , srcInfoPoints = []
                           }
                         (Ident
                            SrcSpanInfo
                              { srcInfoSpan =
                                  SrcSpan "tests/examples/MultipleModules/MixedVersion.hs" 6 10 6 11
                              , srcInfoPoints = []
                              }
                            "y"))))))
          Nothing
      ]
  , []
  )
